zproject/backends.py

```diff
@@ -795,8 +795,7 @@ def is_account_realm_access_forbidden(self, ldap_user: _LDAPUser, realm: Realm)
         # target subdomain matches the user's LDAP attributes.
         realm_access_control = settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL
         if not (
-            isinstance(realm_access_control, dict)
-            and realm.subdomain in realm_access_control
+            realm.subdomain in realm_access_control
```

Comment:
Yeah, looks right to me, this is either `None` or `dict` and the `if` just above this ensures `not None`.

Maybe we should take this opportunity to just extract `realm_access_control = settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL` earlier on in this method and refer to `realm_access_control` rather than repeating `settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL`?

Reply:
That makes sense. Updated the PR.